lib/vyasa/sangh.ex

```diff
@@ -490,10 +490,8 @@ defmodule Vyasa.Sangh do
 
   # way number 3 -- no assoced parent (i.e. will be root sheaf), nothing to reconcile
   def make_reply(
-        %Sheaf{
-          parent_id: nil
-        } = reply,
-        attrs
+        %Sheaf{} = reply,
+        %{parent: nil} = attrs
       ) do
```

Comment:
# [BUG] Odd bug again in sheaf creation @ `make_reply()`

likely need you to add verify this on the changeset side @ks0m1c 

Not sure if the fix is actually something to be done at the changeset
level. In brief, the SOT should be the reply<sub>to</sub> context that
is in the socket state. So even if a sheaf may have a
currently-associated parent sheaf, if the socket says that the
parent<sub>id</sub> or parent is null then it's null (and a root sheaf).

See lib/vyasa<sub>web</sub>/components/contexts/read.ex

``` elixir
def handle_event(
      "sheaf::publish",
      %{
        "body" => body
      } = _params,
      %Socket{
        assigns: %{
          draft_reflector: %Sheaf{} = draft_sheaf,
          draft_reflector_ui: %SheafUiState{
            marks_ui: %MarksUiState{} = _ui_state
          },
          reply_to: reply_to_sheaf,
          session: %VyasaWeb.Session{
            name: username,
            sangh: %Vyasa.Sangh.Session{
              id: sangh_id
            }
          }
        }
      } = socket
    ) do
  IO.inspect(%{body: body},
    label: "SHEAF CREATION without parent"
  )

  payload_precursor = %{
    body: body,
    traits: ["published"],
    signature: username,
    inserted_at: Utils.Time.get_utc_now()
  }

  # FIXME: the socket state should be the SOT. So even if the draft sheaf has an associated parent,
  # when my reply_to_sheaf is nil, then the parent should be set to nil if it gets published.
  # currently this is NOT happening, i.e. the parent assoc is still there, so the sheaf gets published but as a child of whatever the previous state was.
  reply_payload =
    case reply_to_sheaf do
      %Sheaf{} ->
        payload_precursor |> Map.put(:parent, reply_to_sheaf)

      nil ->
        # so the socket's reply_to will take priority, even if the draft sheaf may already an associated parent, this will take priority.
        payload_precursor |> Map.put(:parent, nil)
    end

  draft_sheaf
  |> Vyasa.Sangh.make_reply(reply_payload)

  {:noreply,
   socket
   |> ui_toggle_show_sheaf_modal?()
   |> register_sheaf(Sheaf.draft!(sangh_id))
   |> assign(reply_to: nil)
   |> cascade_stream_change()}
end
```